[zhikaiong2001] iP

zhikaiong2001 opened the pull request:
# Atlas
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
Atlas allows you to remember things easily. It's
* Easy to use
* Interactive

And it is ~~CHEAP~~ **FREE!**

All you need to do is,
1. [Download](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html) 
2. Double-click it
3. Add your tasks
4.  Let it manage your task for you 😉 

Features:
- [X] Manage tasks
- [ ] Reminders (coming soon)
***
If you are a Java Programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
        Atlas atlas = new Atlas();
        atlas.run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

zhikaiong2001 committed:
Level 0. Rename, Greet, Exit

zhikaiong2001 committed:
Level 1. Echo

zhikaiong2001 committed:
Level 2. Add, List

zhikaiong2001 committed:
Added Task class and edited Atlas such as to add the ability to mark tasks as done and undone

zhikaiong2001 committed:
Added 3 kinds of child tasks

zhikaiong2001 committed:
Added automated Text UI Testing, as well as edited the sentence right after adding a task

zhikaiong2001 committed:
Edited automated testing to be more comprehensive

zhikaiong2001 committed:
Added Exception handling
1) marking non-existing task
2) unmarking non-exisitng task
3) invalid format for deadline
4) invalid format for event
5) Invalid command

zhikaiong2001 committed:
Added delete function, replaced array with arraylist, adjusted automated ui testing such as to handle the new command

zhikaiong2001 committed:
Add save and load function for AtlasBot

AtlasBot cannot save data from previous chat

Let's,
* add a method to load tasks from a file when AtlasBot starts, creates a new file/directory if it does not exist
* add a method to save tasks to a file
* add a method to parse each line in the file into a Task

Straight-forward way to implement these functions

zhikaiong2001 committed:
Merge branch 'branch-Level-7'

zhikaiong2001 committed:
Add date and time functionality for AtlasBot using java.time.DataTimeFormatter and java.time.LocalDateTime

zhikaiong2001 committed:
Break down Atlas class into UI, TaskList, Storage, Parser, Command Classes for more oop

zhikaiong2001 committed:
Break Command class into subclasses for each command for more oop.

zhikaiong2001 committed:
Merge branch 'branch-A-MoreOOP'

zhikaiong2001 committed:
Organize project in packages: atlas, atlas.task, atlas.command, atlas.exception

zhikaiong2001 committed:
Merge branch 'add-gradle-support'

zhikaiong2001 committed:
Add JUnit tests for tasks

zhikaiong2001 committed:
Merge branch 'branch-A-JUnit'

zhikaiong2001 committed:
Edit build.gradle file to create jar file

zhikaiong2001 committed:
Add JavaDoc comments to the code

zhikaiong2001 committed:
Reformat code to adhere to coding standards

zhikaiong2001 committed:
Add find functionality to bot

Bot has no function for user to search tasks easily

Let's search the keyword provided by user in the list of tasks

zhikaiong2001 committed:
Merge branch 'branch-A-JavaDoc'

zhikaiong2001 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/atlas/command/AddToDoCommand.java
#	src/main/java/atlas/command/Command.java
#	src/main/java/atlas/command/ListCommand.java

zhikaiong2001 committed:
Merge branch 'branch-Level-9'

zhikaiong2001 committed:
Add JavaDoc comments to FindCommand.java

chewbum commented on src/main/java/atlas/Atlas.java:
```diff
@@ -0,0 +1,56 @@
+ * The main class that drives the Atlas application.
+ */
+public class Atlas {
+    private final String DATA_PATH = "./data/Atlas.txt";
```
I like how your constant name is capitalized and there was the use of underscore to separate words!

chewbum commented on src/main/java/atlas/Parser.java:
```diff
@@ -0,0 +1,136 @@
+package atlas;
+
+import atlas.command.*;
+import atlas.exception.AtlasException;
+import atlas.exception.InvalidDeadlineFormatException;
+import atlas.exception.InvalidEventFormatException;
+import atlas.task.Deadline;
+import atlas.task.Event;
+import atlas.task.Task;
+import atlas.task.ToDo;
+
```
I like how your packages are named! Very intuitive.

chewbum commented on src/main/java/atlas/command/AddToDoCommand.java:
```diff
@@ -0,0 +1,38 @@
+     * Executes the addition of a new ToDo task.
+     */
+    @Override
+    public void execute() {
+        ToDo todo = new ToDo(description);
+        tasks.addTask(todo);
+        ui.showTaskAdded(tasks);
+    }
```
Verbs are used in method naming and they are descriptive enough!

zhikaiong2001 replied:
Thanks for the comments!

zhikaiong2001 committed:
Integrate and use checkStyle to detect coding standard violations

zhikaiong2001 committed:
Merge branch 'branch-A-CheckStyle'

zhikaiong2001 committed:
Add a basic Gui for the AtlasBot using JaxaFX

zhikaiong2001 committed:
Merge branch 'branch-Level-10'

whitesnowx commented on src/main/java/atlas/Ui.java:
```diff
@@ -0,0 +1,164 @@
+        String str = al.get(index).toString();
+        sb.append(str);
+        return sb.toString();
+    }
```
Is this a mistake? 
Might be hard to read since not in proper camelCase

whitesnowx reviewed (commented):
Looks really nice!
Organised and well named

zhikaiong2001 committed:
Edit build.gradle to change build settings

zhikaiong2001 committed:
Add assert statements to ensure important assumptions hold

The code does not have any assert statement

The code makes some assumptions of the program state

The program may not run ideally if the assumptions do not hold true

Let's add assert statements to verify the assumptions

Using assert statements will allow the program to terminate with an error message as soon as the bug is encountered

zhikaiong2001 committed:
Refactor code to adhere to code quality standards as much as possible

Code does not adhere to code quality standards

Code's readibility is poor due to long methods, deep nested and complicated expressions

Let's extract out methods from long methods to adhere to SLAP and break down complicated expressions into smaller ones

Adhering to SLAP and breaking down complex expressions allows for greater readability using abstraction

zhikaiong2001 committed:
Add .yml file into \.github\workflows folder

zhikaiong2001 committed:
Merge pull request #1 from zhikaiong2001/branch-A-Assertions

Add assert statements to ensure important assumptions hold

zhikaiong2001 committed:
Merge branch 'master' of https://github.com/zhikaiong2001/ip

zhikaiong2001 committed:
Merge branch 'master' into branch-A-CodeQuality

zhikaiong2001 committed:
Merge pull request #2 from zhikaiong2001/branch-A-CodeQuality

Refactor code to adhere to code quality standards as much as possible

zhikaiong2001 committed:
Merge branch 'master' of https://github.com/zhikaiong2001/ip

zhikaiong2001 committed:
Merge branch 'branch-A-CI'

zhikaiong2001 committed:
Minor edit to fix bug whereby adding task would lead to failed assert

zhikaiong2001 committed:
Delete .yml file due to build errors

zhikaiong2001 committed:
Add priority functionality to program

Chatbot lacks priority function

It is hard to tell which tasks are of more importance

Let's implement a priority functionality so that when adding new tasks, priority can be specified, as well as a command to update the priority of a pre-existing task

Easy to implement since existing code can be tweaked abit to accomodate an additional keyword

zhikaiong2001 committed:
Merge branch 'branch-BCD-Extension'

zhikaiong2001 committed:
Edit priority so that user can only input between range of 0 to 5, else exception would be thrown

zhikaiong2001 committed:
Add readme and fix minor bugs

zhikaiong2001 committed:
Edit readme and change representative screenshot

zhikaiong2001 committed:
Merge branch 'A-UserGuide'

zhikaiong2001 committed:
Rename picture

zhikaiong2001 committed:
Rename ui.png

zhikaiong2001 committed:
Update README.md

zhikaiong2001 committed:
Update README.md

zhikaiong2001 committed:
Change Ui.png's name in README.md

zhikaiong2001 committed:
Fix a bug where marking a completed task would unmark it

zhikaiong2001 committed:
Update README.md

zhikaiong2001 committed:
Fix bug where deleting non-existent task is allowed

Non-existent task could be deleted

It is not as intended

Let's check the boundaries before allowing deletion

This is to throw an exception incase the specified index is out of bounds

zhikaiong2001 committed:
Edit JUnit tests to work with the current version of code and add missing JavaDoc Comments for public methods

zhikaiong2001 committed:
Update README.md

zhikaiong2001 committed:
Update README.md

zhikaiong2001 committed:
Update README.md

zhikaiong2001 committed:
Update README.md

zhikaiong2001 committed:
Update README.md
